zebra-network/src/peer_set/set.rs

```diff
@@ -393,31 +382,25 @@ where
 
                     self.guards.extend(handles);
 
-                    None
+                    Poll::Ready(Ok(()))
                 }
 
-                // The tasks have been sent and consumed, but then they exited.
-                //
-                // Correctness: the peer set must receive at least one task.
-                //
-                // TODO: refactor `handle_rx` and `guards` into an enum
-                //       for the background task state: Waiting/Running/Shutdown.
```

Comment:
Do we still want this TODO?

Reply:
I'm not sure if it makes sense any more. We might want to create `TaskGuard` and `TaskGuardSet` types, or make what we've done in the state with the update task more generic, and move it to `zebra_chain::task`.

I'm not sure, let's see what's actually needed later.